### ISSUE TITLE
trackpadScrollCausesScale not working as expected on InteractiveViewer

### ISSUE THREAD

**Issue by luis901101** [flutter/flutter#131971]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

When using `trackpadScrollCausesScale: true` in `InteractiveViewer` it doesn't work as expected, the mouse or trackpad scroll gesture doesn't scale/zoom the content, but instead scrolls it.

However, I found something strange, if I touch outside the browser, so that the browser loses focus, and then move the mouse pointer inside the `InteractiveViewer` area and press the **Control** key on the keyboard and do a scroll of the mouse / trackpad, then I do get the expected behavior, that is, the content is scaled.

### Expected results

Content should scale/zoom in/out when mouse/trackpad scrolls.

### Actual results

Content is being scrolled

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

main() {
  runApp(const InteractiveViewerSampleApp());
}

class InteractiveViewerSampleApp extends StatelessWidget {
  const InteractiveViewerSampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('InteractiveViewerSample'),
        ),
        body: const InteractiveViewerSampleWidget(),
      ),
    );
  }
}

class InteractiveViewerSampleWidget extends StatelessWidget {
  const InteractiveViewerSampleWidget({super.key});

  @override
  Widget build(BuildContext context) {
    return Center(
      child: InteractiveViewer(
        constrained: true,
        minScale: 0.1,
        maxScale: 100,
        trackpadScrollCausesScale: true,
        child: Container(
          color: Colors.teal,
          width: 500,
          height: 500,
          alignment: Alignment.center,
          child: const FlutterLogo(),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>
https://github.com/flutter/flutter/assets/28322469/4c0c1311-3d07-4a61-9be5-2343782ea412
</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.0 22A380 darwin-arm64, locale en-US)
    • Flutter version 3.10.6 on channel stable at /Users/user/fvm/versions/3.10.6
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (3 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/user/Android/SDK
    • Platform android-33, build-tools 33.0.1
    • ANDROID_HOME = /Users/user/Android/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
```

</details>

**Comment by ljmatan** [comment]
With GestureDetector I also don't seem to get any effect with this property set to true.

**Comment by darshankawar** [comment]
Thank for the report @luis901101 
A similar issue was fixed and merged into master version recently that you can take a look and see if it helps to resolve your issue or not. https://github.com/flutter/flutter/pull/127114

The fix doesn't seem to be in 3.10 stable version yet.
